linera-execution/src/execution_state_actor.rs

```diff
@@ -141,9 +161,10 @@ where
                     )
                     .await?;
```

Comment:
I'm very scared about holding the `RwLock` guard over an await point. This means that (now as part of the external API of this module) the caller must know all the  possible holders of the lock and make sure to only poll them all at once, since otherwise they will deadlock when taking the lock.